### PR TITLE
[WHISPER] Update modeling tests

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Fixes the test after the update of the tokenizer, which added sufix and prefix tokens.

### CODE REVIEW

**Review by ydshieh — APPROVED** [review]
Thank you @ArthurZucker . LGTM, but let @sanchit-gandhi have a review too, as he is better whisper than me!

File: tests/models/whisper/test_modeling_tf_whisper.py [hunk 1]
```diff
@@ -650,9 +650,9 @@ def _test_large_logits_librispeech(in_queue, out_queue, timeout):
         input_speech = _load_datasamples(1)
 
         processor = WhisperProcessor.from_pretrained("openai/whisper-large")
-        processed_inputs = processor(audio=input_speech, text="This part of the speech", return_tensors="tf")
+        processed_inputs = processor(audio=input_speech, return_tensors="tf")
         input_features = processed_inputs.input_features
-        labels = processed_inputs.labels
+        labels = tf.convert_to_tensor([[5723, 644, 295, 220, 3322, 6218]])
```
Comment (ydshieh) [inline tests/models/whisper/test_modeling_tf_whisper.py:655]: could you explain why we don't pass `text`, and use manual `labels`?

**Review by ydshieh — CHANGES_REQUESTED** [review]
Sorry, I have a question now 🙏 !

File: tests/models/whisper/test_modeling_tf_whisper.py (hunk 1 above)
Comment (ArthurZucker) [reply tests/models/whisper/test_modeling_tf_whisper.py:655]: Basically it’s just simpler to remove the prefix and suffix ☺️

Comment (ydshieh) [reply tests/models/whisper/test_modeling_tf_whisper.py:655]: Got it.

If `removing prefix and suffix` is the necessary step to get the correct labels, when a user use the Whisper model, they has to do this manually, and cannot just 
```python
processed_inputs = processor(audio=input_speech, text="This part of the speech", return_tensors="tf")
labels = processed_inputs.labels
```
? I think the above code block should prepare the correct labels to be used for the models, otherwise we will get a lot of questions from the users in the future.

cc @sanchit-gandhi and @sgugger

Comment (ArthurZucker) [reply tests/models/whisper/test_modeling_tf_whisper.py:655]: Well the above code is correct! But for testing, we used a special prompt, without the tokens that are necessary for the actual use of the model!🤗 
Overall, it's cleaner to just keep the logits and more resilient to changes in the tokenizer!

Comment (ydshieh) [reply tests/models/whisper/test_modeling_tf_whisper.py:655]: Ah, it is used as `decoder_input_ids`!

File: tests/models/whisper/test_modeling_whisper.py [hunk 2]
```diff
@@ -853,9 +853,9 @@ def test_large_logits_librispeech(self):
         input_speech = self._load_datasamples(1)
 
         processor = WhisperProcessor.from_pretrained("openai/whisper-large")
-        processed_inputs = processor(audio=input_speech, text="This part of the speech", return_tensors="pt")
+        processed_inputs = processor(audio=input_speech, return_tensors="pt")
```
Comment (sanchit-gandhi) [inline tests/models/whisper/test_modeling_whisper.py:856]: Would personally prefer that we don't hardcode the labels and instead tokenise the target text. We can set `add_special_tokens` to False if we don't want the 'special' BOS and EOS tokens, as we do here:
https://github.com/huggingface/transformers/blob/cbbeca3d1733aa7c9b443af5ff231a5affcd8a1e/tests/models/whisper/test_tokenization_whisper.py#L120

```suggestion
        processed_inputs = processor(audio=input_speech, text="This part of the speech", add_special_tokens=False, return_tensors="pt")
```

File: tests/models/whisper/test_modeling_tf_whisper.py [hunk 3]
```diff
@@ -650,9 +650,9 @@ def _test_large_logits_librispeech(in_queue, out_queue, timeout):
         input_speech = _load_datasamples(1)
 
         processor = WhisperProcessor.from_pretrained("openai/whisper-large")
-        processed_inputs = processor(audio=input_speech, text="This part of the speech", return_tensors="tf")
+        processed_inputs = processor(audio=input_speech, return_tensors="tf")
```
Comment (sanchit-gandhi) [inline tests/models/whisper/test_modeling_tf_whisper.py:653]: Same here for TF if possible!

File: tests/models/whisper/test_modeling_tf_whisper.py (hunk 1 above)
Comment (sanchit-gandhi) [reply tests/models/whisper/test_modeling_tf_whisper.py:655]: I'm in agreement with @ydshieh that we should use the text if possible (over hardcoding the labels).

**Review by sanchit-gandhi — COMMENTED** [review]
Thanks for the fix @ArthurZucker! The slow tokenizer integration test looks good - have some comments regarding the logits test (see below).

**Comment by ArthurZucker** [comment]
I agree, it is indeed better to just use `add_special_tokens=False`! Slipped my mind 😉

**Comment by ydshieh** [comment]
Ping me again once when you think it's ready :)

**Comment by sanchit-gandhi** [comment]
Looks like we still need to add `add_special_tokens=False` to the tf test!

**Comment by ArthurZucker** [comment]
Yeah on it 🤗

**Comment by ArthurZucker** [comment]
Ah there's small bug with this, the kwargs is passed to `self.pad` in the feature extractor. Gonna fix that

File: src/transformers/models/whisper/feature_extraction_whisper.py [hunk 4]
```diff
@@ -307,7 +307,6 @@ def __call__(
             max_length=max_length if max_length else self.n_samples,
             truncation=truncation,
             pad_to_multiple_of=pad_to_multiple_of,
-            **kwargs,
```
Comment (sanchit-gandhi) [inline src/transformers/models/whisper/feature_extraction_whisper.py:310]: Was this wrong before? If we don't use the kwargs we can remove them from the call in L288:
https://github.com/huggingface/transformers/blob/d24e84d9edde889bd56ccd46617208cb118d66fa/src/transformers/models/whisper/feature_extraction_whisper.py#L228

**Review by sanchit-gandhi — APPROVED** [review]
Thanks for iterating @ArthurZucker! Just one question about the removal of the `kwargs` in the feature extractor call method (see below).

File: tests/models/whisper/test_modeling_tf_whisper.py [hunk 5]
```diff
@@ -650,13 +650,15 @@ def _test_large_logits_librispeech(in_queue, out_queue, timeout):
         input_speech = _load_datasamples(1)
 
         processor = WhisperProcessor.from_pretrained("openai/whisper-large")
-        processed_inputs = processor(audio=input_speech, text="This part of the speech", return_tensors="tf")
+        processed_inputs = processor(
+            audio=input_speech, text="This part of the speech", add_special_tokens=False, return_tensors="tf"
+        )
         input_features = processed_inputs.input_features
-        labels = processed_inputs.labels
+        decoder_input_ids = processed_inputs.labels
```
Comment (ydshieh) [inline tests/models/whisper/test_modeling_tf_whisper.py:657]: I remembered in a previous commit, you need to use a hard-coded tensor, which is the labels with some tokens removed.

Here the name `decoder_input_ids` is good, but the value is just `processed_inputs.labels`.

I am somehow confused by if we should have some tokens removed from labels or not, or it doesn't matter at all, and both work? (I will be shocked if this is the case)

File: src/transformers/models/whisper/feature_extraction_whisper.py (hunk 4 above)
Comment (ArthurZucker) [reply src/transformers/models/whisper/feature_extraction_whisper.py:310]: Yes, the `add_special_tokens` was passed down until here. Checked that `wav2vec2` does not use `kwargs` for padding, plus the call is similar and `kwargs` are not popped out, so appears as the good solution.

File: tests/models/whisper/test_modeling_tf_whisper.py (hunk 5 above)
Comment (ArthurZucker) [reply tests/models/whisper/test_modeling_tf_whisper.py:657]: Haha feel free to read the suggestions add by @sanchit-gandhi but basically, it gives the same result. I hard coded to remove the `special_tokens`, and passing `add_special_tokens=False` is enough to remove the special tokens that where failing the tests.

File: src/transformers/models/whisper/feature_extraction_whisper.py (hunk 4 above)
Comment (ArthurZucker) [reply src/transformers/models/whisper/feature_extraction_whisper.py:310]: We might have to keep them for pipeline compatibility, will check that thanks 👍🏻

**Review by ydshieh — APPROVED** [review]
LGTM now! Thank you for your patience. Will leave @sanchit-gandhi and you to final the PR regarding the kwargs

File: src/transformers/models/whisper/feature_extraction_whisper.py (hunk 4 above)
Comment (sanchit-gandhi) [reply src/transformers/models/whisper/feature_extraction_whisper.py:310]: Okay brill - sounds good in that case (pop from here, **not** in L228).

File: tests/models/whisper/test_modeling_tf_whisper.py (hunk 5 above)
Comment (ydshieh) [reply tests/models/whisper/test_modeling_tf_whisper.py:657]: Yes, I understand now with all the explanation :-) Thanks.
